index.d.ts

```diff
@@ -238,7 +238,9 @@ declare module '@primer/components' {
 
   export interface FormGroupProps extends CommonProps, Omit<React.HTMLAttributes<HTMLDivElement>, 'color'> {}
 
-  export interface FormGroupLabelProps extends CommonProps, TypographyProps, Omit<React.HTMLAttributes<HTMLLabelElement>, 'color'> {}
+  export interface FormGroupLabelProps extends CommonProps, TypographyProps, Omit<React.HTMLAttributes<HTMLLabelElement>, 'color'> {
+    htmlFor?: string
+  }
```

Comment:
Hmm, it's strange that `React.HTMLAttributes<HTMLLabelElement>` doesn't include `htmlFor`.  `React.ComponentPropsWithRef<'label'>` should do the trick though. I recently learned that `ComponentPropsWithRef` is the preferred way to get props of HTML elements anyway ([link](ComponentPropsWithRef)). 
```suggestion
  export interface FormGroupLabelProps extends CommonProps, TypographyProps, Omit<React.ComponentPropsWithRef<'label'>, 'color'> {}
```

Reply:
Oops, I just tried this locally and I'm getting errors. I'm fine with merging the PR as-is since we're about to replace this whole `index.d.ts` file very soon.

Reply:
For extra context, I see the `@types/react` version of `LabelHTMLAttributes` has the field:

```ts
    interface LabelHTMLAttributes<T> extends HTMLAttributes<T> {
        form?: string;
        htmlFor?: string;
    }
```

However `HTMLLabelElement` in `@types/react` doesn't seem to have this information:

```ts
interface HTMLLabelElement extends HTMLElement { }
```

I think this would be the fix without duplicating `htmlFor` but that feels like a big hammer given the refactorings underway:

```ts
export interface FormGroupLabelProps extends CommonProps, TypographyProps, Omit<LabelHTMLAttributes, 'color'> {}
```